src/exception_utils.rs

```diff
@@ -230,85 +230,32 @@ pub fn exception_data_abort_access_is_sign_ext() -> bool {
     ((exception_iss() >> 21) & 1) != 0
 }
 
-/// Macro to save the general-purpose registers (GPRs) to the stack.
-///
-/// This macro generates assembly code that:
-/// - Subtracts the size of the register data from the stack pointer.
-/// - Stores all 31 general-purpose registers (x0 to x30) on the stack.
-/// - Saves the `elr_el2` and `spsr_el2` registers on the stack as well.
-/// - Adjusts the stack pointer after storing the registers.
-///
-/// The layout of the saved registers on the stack is:
-/// - Registers x0 to x29 at `sp` to `sp + 29 * 8`.
-/// - Registers x30 and the stack pointer at `sp + 30 * 8`.
-/// - Registers `elr_el2` and `spsr_el2` at `sp + 32 * 8`.
+/// Macro to save the host ctx to the stack.
 macro_rules! save_regs_to_stack {
     () => {
         "
-        sub     sp, sp, 34 * 8
-        stp     x0, x1, [sp]
-        stp     x2, x3, [sp, 2 * 8]
-        stp     x4, x5, [sp, 4 * 8]
-        stp     x6, x7, [sp, 6 * 8]
-        stp     x8, x9, [sp, 8 * 8]
-        stp     x10, x11, [sp, 10 * 8]
-        stp     x12, x13, [sp, 12 * 8]
-        stp     x14, x15, [sp, 14 * 8]
-        stp     x16, x17, [sp, 16 * 8]
-        stp     x18, x19, [sp, 18 * 8]
-        stp     x20, x21, [sp, 20 * 8]
-        stp     x22, x23, [sp, 22 * 8]
-        stp     x24, x25, [sp, 24 * 8]
-        stp     x26, x27, [sp, 26 * 8]
-        stp     x28, x29, [sp, 28 * 8]
-
-        mov     x1, sp
-        add     x1, x1, #(0x110)
-        stp     x30, x1, [sp, 30 * 8]
-        mrs     x10, elr_el2
-        mrs     x11, spsr_el2
-        stp     x10, x11, [sp, 32 * 8]
-
-        add    sp, sp, 34 * 8"
+        sub     sp, sp, 12 * 8
+        stp     x29, x30, [sp, 10 * 8]
+        stp     x27, x28, [sp, 8 * 8]
+        stp     x25, x26, [sp, 6 * 8]
+        stp     x23, x24, [sp, 4 * 8]
+        stp     x21, x22, [sp, 2 * 8]
+        stp     x19, x20, [sp]
+        add    sp, sp, 12 * 8"
     };
 }
 
-/// Macro to restore the general-purpose registers (GPRs) from the stack.
-///
-/// This macro generates assembly code that:
-/// - Subtracts the size of the register data from the stack pointer.
-/// - Restores all 31 general-purpose registers (x0 to x30) from the stack.
-/// - Loads the `elr_el2` and `spsr_el2` registers from the stack.
-/// - Adjusts the stack pointer after restoring the registers.
-///
-/// The layout of the restored registers on the stack matches the layout
-/// defined in `save_regs_to_stack!`.
+/// Macro to restore the host ctx from the stack.
```

Comment:
expand the comments please, not just simply delete them...

Reply:
I think the comments here is enough, a aarch64er who force on exception handling needn't more detail explanations.

Reply:
At least you need to explain what is `host ctx`